src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingExtractor.java

```diff
@@ -60,13 +62,17 @@ Hearing extractHearing(String payload, String messageId) {
         }
     }
 
-    private Hearing parseCPMessage(SnsMessageContainer snsMessageContainer) throws JsonProcessingException {
+    private List<Hearing> parseCPMessage(SnsMessageContainer snsMessageContainer) throws JsonProcessingException {
         final var cpHearingEvent = commonPlatformParser.parseMessage(snsMessageContainer.getMessage(), CPHearingEvent.class);
         final var hearing = cpHearingEvent.asDomain();
-        return passHearingIdToCourtCaseService ? hearing
-                .withHearingId(cpHearingEvent.getHearing().getId())
-                .withHearingEventType(snsMessageContainer.getHearingEventType().getValue())
-                : hearing;
+        return passHearingIdToCourtCaseService ? setHearingAttributes(hearing, cpHearingEvent, snsMessageContainer) : hearing;
```

Comment:
[This condition is always true](https://github.com/search?q=repo%3Aministryofjustice%2Fcourt-case-matcher%20pass-hearing-id-to-court-case-service&type=code) as far as I can see - we could take the opportunity to do a small PR to remove it.